src/wayland/tablet_manager/tablet.rs

```diff
@@ -0,0 +1,98 @@
+use std::ops::Deref as _;
+use std::path::PathBuf;
+use std::{cell::RefCell, rc::Rc};
+
+use wayland_protocols::unstable::tablet::v2::server::zwp_tablet_seat_v2::ZwpTabletSeatV2;
+use wayland_protocols::unstable::tablet::v2::server::zwp_tablet_v2::ZwpTabletV2;
+use wayland_server::protocol::wl_surface::WlSurface;
+use wayland_server::Filter;
+
+use crate::backend::input::Device;
+
+/// Description of graphics tablet device
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub struct TabletDescriptor {
+    /// Tablet device name
+    pub name: String,
+    /// Tablet device USB product id
+    pub id_product: Option<u32>,
+    /// Tablet device USB vendor id
+    pub id_vendor: Option<u32>,
+    /// Path to the device
+    pub syspath: Option<PathBuf>,
+}
+
+impl<D: Device> From<D> for TabletDescriptor {
```

Comment:
Semantically, wouldn't it be more precise for it to be `impl<D: Device> From<&D> for ...` ? That impl does not need to consume the device.